src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfig/IEditorConfigStorageLocation2.cs

```diff
@@ -6,17 +6,13 @@
 
 #if CODE_STYLE
 using OptionSet = Microsoft.CodeAnalysis.Diagnostics.AnalyzerConfigOptions;
```

Comment:
NOTE: The only real `#if CODE_STYLE` statements that should remain after this PR are to distinguish between `AnalyzerConfigOptions` and `OptionSet` in CodeStyle and Workspaces layer respectively. Given Sam's recent support for `OptionSet.AsAnalyzerConfigOptions`, it shouldn't be very hard to even get rid of these in a follow-up PR.

Reply:
`IEditorConfigStorageLocation2` is one case where we absolutely can't remove the distinction. This interface is used to convert `OptionSet` to .editorconfig format; if you already have the latter then the whole interface doesn't make sense.

Reply:
Sounds good - I think these exception cases are fine.